src/GovUk.Education.ExploreEducationStatistics.Publisher.Tests/Services/NotificationsServiceTests.cs

```diff
@@ -0,0 +1,222 @@
+                        new()
+                        {
+                            Name = release1.Publication.Title,
+                            PublicationId = release1.Publication.Id,
+                            Slug = release1.Publication.Slug,
+                        },
+                        new()
+                        {
+                            Name = release3.Publication.Title,
+                            PublicationId = release3.Publication.Id,
+                            Slug = release3.Publication.Slug,
+                        },
+                    }))), Times.Once);
+            }
+
+            MockUtils.VerifyAllMocks(storageQueueService);
+        }
+
+        [Fact]
+        public async Task NotifySubscribersIfApplicable_NotifyOnSet()
+        {
+            var release = new Release
+            {
+                Id = Guid.NewGuid(),
+                Publication = new Publication
+                {
+                    Id = Guid.NewGuid(),
+                    Title = "pub title",
+                    Slug = "pub-slug",
+                },
+                ReleaseStatuses = new List<ReleaseStatus>
+                {
+                    new()
+                    {
+                        Id = Guid.NewGuid(),
+                        ApprovalStatus = ReleaseApprovalStatus.Approved,
+                        Created = DateTime.UtcNow,
+                        NotifySubscribers = true,
+                        NotifiedOn = DateTime.UtcNow,
+                    },
+                },
+            };
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddRangeAsync(release);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var storageQueueService = new Mock<IStorageQueueService>(MockBehavior.Strict);
+            storageQueueService.Setup(mock => mock.AddMessages(
+                PublicationQueue,
+                It.IsAny<List<PublicationNotificationMessage>>()))
+                .Returns(Task.CompletedTask);
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var notificationsService = BuildNotificationsService(
+                    contentDbContext, storageQueueService.Object);
+
+                await notificationsService.NotifySubscribersIfApplicable(release.Id);
+
+                storageQueueService.Verify(mock => mock.AddMessages(
+                    PublicationQueue,
+                It.IsAny<List<PublicationNotificationMessage>>()), Times.Never);
+            }
+        }
+
+        [Fact]
+        public async Task NotifySubscribersIfApplicable_None()
+        {
+            var storageQueueService = new Mock<IStorageQueueService>(MockBehavior.Strict);
+            storageQueueService.Setup(mock => mock.AddMessages(
```

Comment:
I'm not sure what the purpose of this setup is because it should never be called if there's no release id's.

I would simplify this massively in combination with my other suggested to create the default queue service mock with `Strict` behaviour when building the service.

```
        [Fact]
        public async Task NotifySubscribersIfApplicable_NoReleaseIds()
        {
            await using (var contentDbContext = InMemoryContentDbContext())
            {
                var notificationsService = BuildNotificationsService(contentDbContext);

                await notificationsService.NotifySubscribersIfApplicable();

                // No interaction with the queue service is expected
                // since no releases are being published
            }
        }
```

Reply:
Seems I was confused about what was necessary to check that a method didn't get called - fixed.